suite-native/graph/src/components/TransactionEvent.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { runOnJS, useDerivedValue, withSpring } from 'react-native-reanimated';
+
+import { Circle, Group } from '@shopify/react-native-skia';
+
+import { EventComponentProps } from '@suite-native/react-native-graph/src/LineGraphProps';
+import { Color } from '@trezor/theme';
+import { useNativeStyles } from '@trezor/styles';
+import { GroupedBalanceMovement } from '@suite-common/graph';
+
+import { GraphContextProvider } from './GraphContextProvider';
+import { TransactionEventTooltipProps } from './TransactionEventTooltip';
+
+type EventVariant = 'positive' | 'negative';
+
+type TransactionEventProps = EventComponentProps &
+    GroupedBalanceMovement['payload'] & {
+        handleActiveTransactionEvent: (
+            newValue: TransactionEventTooltipProps | null,
+            cords: { eventX: number; eventY: number },
+        ) => void;
+    };
+
+const INNER_DOT_WIDTH = 2.5;
+const ACTIVE_INNER_DOT_WIDTH = 4.5;
+const OUTER_DOT_WIDTH = 6;
+const ACTIVE_OUTER_DOT_WIDTH = 8;
+
+const variantToStylesMaps = {
+    positive: {
+        innerColor: 'backgroundPrimaryDefault',
+        outerColor: 'backgroundPrimarySubtleOnElevation0',
+    },
+    negative: {
+        innerColor: 'backgroundAlertRedBold',
+        outerColor: 'backgroundAlertRedSubtleOnElevation0',
+    },
+} as const satisfies Record<EventVariant, { innerColor: Color; outerColor: Color }>;
+
+const TransactionEventContent = ({
```

Comment:
I'd probably wrap this in memo

Reply:
Not sure about that, any specific reason?